### PR TITLE
[DOCS] Bump Sphinx and Pydata versions

### PR DESCRIPTION
**Pull request by bbielawx**
Sphinx -> 7.2.6
PyData -> 0.14.4

### CODE REVIEW

**Review by akopytko — APPROVED** [review]

File: docs/sphinx_setup/_static/download/OV_2023_models_supported.pdf
Comment (ilya-lavrenov) [inline docs/sphinx_setup/_static/download/OV_2023_models_supported.pdf:1]: do we need to store PDF files here?
if yes, we can put them under LFS?

File: tools/pot/openvino/tools/pot/version.txt
Comment (ilya-lavrenov) [inline tools/pot/openvino/tools/pot/version.txt:1]: please, remove this file

File: docs/sphinx_setup/sphinx.log [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+Traceback (most recent call last):
+  File "C:\Users\bbielawx\OneDrive - Intel Corporation\Desktop\OpenVINO\venv\Lib\site-packages\sphinx\cmd\build.py", line 293, in build_main
+    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  File "C:\Users\bbielawx\OneDrive - Intel Corporation\Desktop\OpenVINO\venv\Lib\site-packages\sphinx\application.py", line 157, in __init__
+    raise ApplicationError(__('Cannot find source directory (%s)') %
+sphinx.errors.ApplicationError: Cannot find source directory (C:\Users\bbielawx\OneDrive - Intel Corporation\Desktop\OpenVINO\docs\sphinx_setup\docs\sphinx_setup)
+
+[91mApplication error:[39;49;00m
+Cannot find source directory (C:\Users\bbielawx\OneDrive - Intel Corporation\Desktop\OpenVINO\docs\sphinx_setup\docs\sphinx_setup)
```
Comment (ilya-lavrenov) [inline docs/sphinx_setup/sphinx.log:11]: I suppose we need to remove this file

Comment (bbielawx) [reply docs/sphinx_setup/sphinx.log]: sphinx.log deleted

File: tools/pot/openvino/tools/pot/version.txt
Comment (bbielawx) [reply tools/pot/openvino/tools/pot/version.txt:1]: version.txt deleted

File: docs/sphinx_setup/_static/download/OV_2023_models_supported.pdf
Comment (bbielawx) [reply docs/sphinx_setup/_static/download/OV_2023_models_supported.pdf:1]: Removed both models_supported pdf files.

File: docs/sphinx_setup/_static/conformance_reports/opset_report_omz_dynamic.html
Comment (kblaszczak-intel) [inline docs/sphinx_setup/_static/conformance_reports/opset_report_omz_dynamic.html:1]: this file can stay without changes, in the download folder

File: docs/sphinx_setup/_static/download/operation_conformance_table_files/opset_report_omz_static.html
Comment (kblaszczak-intel) [inline docs/sphinx_setup/_static/download/operation_conformance_table_files/opset_report_omz_static.html:1]: This change may also be removed, all "conformance table" files should stay in the "conformance_reports" folder under the "download" directory.

File: docs/sphinx_setup/_static/css/benchmark-banner.css
Comment (kblaszczak-intel) [inline docs/sphinx_setup/_static/css/benchmark-banner.css:1]: This change may be removed - we still need this css

**Pull request merged**